src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,148 @@
+  }
+
+  @Override
+  public void setLeftRollerPower(double percentage) {
+    leftRollerMotor.setControl(new DutyCycleOut(percentage));
+  }
+
+  @Override
+  public void setDrumPower(double percentage) {
+    drumMotor.setControl(new DutyCycleOut(percentage));
+  }
+
+  @Override
+  public void setRightRollerCurrent(double current) {
+    rightRollerMotor.setControl(new TorqueCurrentFOC(current));
+  }
+
+  @Override
+  public void setLeftRollerCurrent(double current) {
+    leftRollerMotor.setControl(new TorqueCurrentFOC(current));
+  }
+
+  @Override
+  public void setDrumCurrent(double current) {
+    drumMotor.setControl(new TorqueCurrentFOC(current));
+  }
+
+  private void configureIntakeMotors() {
+    rightRollerMotor = new TalonFX(IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_ID);
+    leftRollerMotor = new TalonFX(IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_ID);
+    drumMotor = new TalonFX(IntakeConstants.INTAKE_DRUM_MOTOR_ID);
+
+    TalonFXConfiguration rightRollerConfig = new TalonFXConfiguration();
+    TalonFXConfiguration leftRollerConfig = new TalonFXConfiguration();
+    CurrentLimitsConfigs rollerCurrentLimits = new CurrentLimitsConfigs();
+
+    rollerCurrentLimits.SupplyCurrentLimit = IntakeConstants.ROLLERS_CONTINUOUS_CURRENT_LIMIT;
+    rollerCurrentLimits.SupplyCurrentThreshold = IntakeConstants.ROLLERS_PEAK_CURRENT_LIMIT;
+    rollerCurrentLimits.SupplyTimeThreshold = IntakeConstants.ROLLERS_PEAK_CURRENT_DURATION;
+    rollerCurrentLimits.SupplyCurrentLimitEnable = true;
+
+    rightRollerConfig.CurrentLimits = rollerCurrentLimits;
+    leftRollerConfig.CurrentLimits = rollerCurrentLimits;
+
+    rightRollerConfig.MotorOutput.Inverted =
+        IntakeConstants.ROLLERS_MOTOR_INVERTED
+            ? InvertedValue.Clockwise_Positive
+            : InvertedValue.CounterClockwise_Positive;
+
+    leftRollerConfig.MotorOutput.Inverted =
+        IntakeConstants.ROLLERS_MOTOR_INVERTED
+            ? InvertedValue.Clockwise_Positive
+            : InvertedValue.CounterClockwise_Positive;
+
+    rightRollerMotor.getConfigurator().apply(rightRollerConfig);
+    leftRollerMotor.getConfigurator().apply(leftRollerConfig);
+
+
+    TalonFXConfiguration drumConfig = new TalonFXConfiguration();
+    CurrentLimitsConfigs drumCurrentLimits = new CurrentLimitsConfigs();
+
+    drumCurrentLimits.SupplyCurrentLimit = IntakeConstants.DRUM_CONTINUOUS_CURRENT_LIMIT;
+    drumCurrentLimits.SupplyCurrentThreshold = IntakeConstants.DRUM_PEAK_CURRENT_LIMIT;
+    drumCurrentLimits.SupplyTimeThreshold = IntakeConstants.DRUM_PEAK_CURRENT_DURATION;
+    drumCurrentLimits.SupplyCurrentLimitEnable = true;
+
+    drumConfig.CurrentLimits = drumCurrentLimits;
+
+    drumConfig.MotorOutput.Inverted =
+        IntakeConstants.DRUM_MOTOR_INVERTED
+            ? InvertedValue.Clockwise_Positive
+            : InvertedValue.CounterClockwise_Positive;
+
+    drumMotor.getConfigurator().apply(drumConfig);
+    
+  }
```

Comment:
Also instantiate the objects for the signals and control requests that are stored in instance variables.